src/Init/Lean/Elab/TermApp.lean

```diff
@@ -375,28 +369,17 @@ else
   else
     mergeFailures candidates f
 
-private partial def expandAppAux : Syntax → Array Syntax → Syntax × Array Syntax
-| stx, args => stx.ifNodeKind `Lean.Parser.Term.app
-  (fun node =>
-    let fn  := node.getArg 0;
-    let arg := node.getArg 1;
-    expandAppAux fn (args.push arg))
-  (fun _ => (stx, args.reverse))
-
-private def expandApp (stx : Syntax) : TermElabM (Syntax × Array NamedArg × Array Arg) := do
-let (f, args) := expandAppAux stx #[];
-(namedArgs, args) ← args.foldlM
-  (fun (acc : Array NamedArg × Array Arg) arg =>
-    let (namedArgs, args) := acc;
-    arg.ifNodeKind `Lean.Parser.Term.namedArgument
-      (fun argNode => do
-        -- `(` ident `:=` term `)`
-        namedArgs ← addNamedArg arg acc.1 { name := argNode.getIdAt 1, val := Arg.stx $ argNode.getArg 3 };
-        pure (namedArgs, args))
-      (fun _ =>
-        pure (namedArgs, args.push $ Arg.stx arg)))
-  (#[], #[]);
-pure (f, namedArgs, args)
+private partial def expandApp : Syntax → TermElabM (Syntax × Array NamedArg × Array Arg)
+| stx => match_syntax stx with
+  | `($fn ($id := $arg)) => do
+    (stx, namedArgs, args) ← expandApp fn;
+    namedArgs ← addNamedArg id namedArgs { name := id.getId, val := Arg.stx arg };
```

Comment:
minor position info change: the ref is now `id`, not the whole namedApp. This seemed reasonable to me.